Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp

```diff
@@ -1950,8 +1967,12 @@ void ResourceLoadStatisticsStore::logCrossSiteLoadWithLinkDecoration(const Navig
     }
     insertDomainRelationshipList(topFrameLinkDecorationsFromQuery, HashSet<RegistrableDomain>({ fromDomain }), *toDomainResult.second);
 
+    auto currentDataRemovalFrequency = dataRemovalFrequency(toDomain);
+    if (currentDataRemovalFrequency == DataRemovalFrequency::Short)
```

Comment:
We should also early return if `currentDataRemovalFrequency == newDataRemovalFrequency`, so next database operations can be skipped.